Adding trigger fields...

jworkman opened the pull request:
> Added the ability to only trigger MC submissions if specific forms fields were not blank, empty, or null. This adds flexability to your Grav forms. Use case would be optional newsletter signup in a form. Only submitting to MC if the user selects opt in.

A specific use case for this feature would be a contact form with a "Sign me up for news & special offers." The MC form only needs to be called when that checkbox is selected.

Commit:
Added the ability to only trigger MC submissions if specific forms fields were not blank, empty, or null. This adds flexability to your Grav forms. Use case would be optional newsletter signup in a form. Only submitting to MC if the user selects opt in.

aaronhipple commented on mailchimp.php:
```diff
@@ -126,11 +126,44 @@ public static function supportedLanguages()
+    /**
+     * @param Event $event
+     */
+    protected function checkTriggers($event)
```
Could we call this something like `shouldSubscribe` and move the declaration below `handleSubscribe`?

- `Subscribe` is the operative verb here, so connecting the method names semantically makes sense to me.
- `handleSubscribe` is the *primary* action, with `checkTriggers`/`shouldSubscribe` being a subsidiary method. As such, I'd declare it below, but still adjacent to the main action.

aaronhipple reviewed (changes requested):
I like! Just one little nitpick about semantics in the function name.

- Could you format this to PSR-2 standards? `phpcs --standard=PSR2 mailchimp.php` and clean up any stragglers, like?
- Mind adding an example of `required_fields` to the example in the README?

Commit:
- Updated checkTriggers subsidiary method to be shouldSubscribe.

- Updated shouldSubscribe method to be PSR-2 compliant.

- Moved shouldSubscribe subsidiary method below primary handleSubscribe method declaration.

- Added README documentation on new required fields feature.

jworkman commented:
- Updated checkTriggers subsidiary method to be shouldSubscribe.
- Updated shouldSubscribe method to be PSR-2 compliant.
- Moved shouldSubscribe subsidiary method below primary handleSubscribe method declaration.
- Added README documentation on new required fields feature.
